### DISCUSSION TITLE
Async underused?

### DISCUSSION THREAD

**Discussion by ghost** [elementary/code#1333] [Q&A]
Hi, I was looking at this issue, trying to find a solution: https://github.com/elementary/code/issues/1332.

~~I came to the conclusion that maybe it was synchronous programming somewhere.~~

~~I'm a beginner to Vala, so this question may be stupid. This function over here is supposed to be async, but there's no `async` keyword.~~

~~https://github.com/elementary/code/blob/f74056da4400e52444b0665ce1cf5b0be8ef86c8/src/Services/FileHandler.vala#LL56C1-L56C73~~

~~This over here uses the sync variant...but should it?~~

~~https://github.com/elementary/code/blob/f74056da4400e52444b0665ce1cf5b0be8ef86c8/src/Services/TemplateManager.vala#L60~~

**Comment by davidmhewitt** [comment]
> I'm a beginner to Vala, so this question may be stupid. This function over here is supposed to be async, but there's no `async` keyword.

What do you mean there's no async keyword? It's right there on the line you've highlighted? And then the code within the function uses `yield` to yield execution to other parts of the code when it's waiting on IO.

> This over here uses the sync variant...but should it?

Maybe it could be re-written to use the async variant, but this code is related to project templates, which isn't related to the sidebar slowness in the issue report.

**Reply by ghost** [reply]
> What do you mean there's no async keyword? It's right there on the line you've highlighted? And then the code within the function uses yield to yield execution to other parts of the code when it's waiting on IO.

I'm so sorry!!! Will keep looking for the problematic code...

**Comment by ghost** [comment]
This doesn't have a return value, I believe making it async would be beneficial:

https://github.com/elementary/code/blob/f74056da4400e52444b0665ce1cf5b0be8ef86c8/src/FolderManager/FileView.vala#L250

What do you think?

**Comment by jeremypw** [comment]
Also need to check if the time taken increases in proportion the number of items or to its square or log; this could give a clue as to the cause. E.g. is the list being resorted after every addition? Is there a list that is being iterated over for every addition?  In the issue, the 2000 files I was loading were empty and being loaded from SSD.
